### PR TITLE
[feature](HiveCatalog) Support for getting hive meta data from relational databases under HMS

### PR DESCRIPTION
**Pull request by BePPPower**
## Proposed changes

Issue Number: close #xxx

**description:**
HiveCatalog needs to get Hive metadata information, and the past HiveCatalog only supports to obtain metadata from HMS.
This pr supports getting Hive metadata information from relational databases under HMS (this pr only supports postgreSQL database)

**Usage**：
We can control whether HiveCatalog gets metadata from HMS or relational database by `hive_meta_type`. The default is hms.
When we set `"hive.metastore.type" = "jdbc"`, HiveCatalog retrieves Hive metadata information from the relational database

**example**:
```SQL
-- get hive metadata information from the relational database
create catalog if not exists docker_hive properties (
    'type'='hms',
    'hive.metastore.type' = 'jdbc',  -- must be jdbc
    "user"="hive",
    "password"="xxxx",
    "jdbc_url" = "jdbc:postgresql://ip:port/metastore?useSSL=false",
    "driver_url" = "https://xxxxxx/postgresql-42.5.0.jar",
    "driver_class" = "org.postgresql.Driver"
);

-- get hive metadata information from HMS
create catalog if not exists docker_hive properties (
    'type'='hms',
    'hive.metastore.type' = 'hms',    -- can be ignored
    'hive.metastore.uris' = 'thrift://172.21.0.101:9083'
);
```

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by BePPPower** [comment]
run buildall

**Comment by BePPPower** [comment]
run buildall

**Comment by BePPPower** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/datasource/hive/HiveJdbcCachedClient.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+// under the License.
+
+package org.apache.doris.datasource.hive;
+
+import org.apache.doris.datasource.jdbc.client.JdbcClient;
+import org.apache.doris.datasource.jdbc.client.JdbcClientConfig;
+
+import com.google.common.base.Preconditions;
+
+public abstract class HiveJdbcCachedClient extends JdbcClient implements CachedClient {
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/datasource/hive/HiveJdbcCachedClient.java:25]:
```suggestion
public abstract class JdbcHMSCachedClient extends JdbcClient implements CachedClient {
```

File: fe/fe-core/src/main/java/org/apache/doris/datasource/hive/CachedClientFactory.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+package org.apache.doris.datasource.hive;
+
+import org.apache.doris.catalog.JdbcResource;
+import org.apache.doris.datasource.jdbc.client.JdbcClient;
+import org.apache.doris.datasource.jdbc.client.JdbcClientConfig;
+
+import com.google.common.base.Preconditions;
+import org.apache.hadoop.hive.conf.HiveConf;
+
+public class CachedClientFactory {
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/datasource/hive/CachedClientFactory.java:27]:
```suggestion
public class HMSCachedClientFactory {
```

File: fe/fe-core/src/main/java/org/apache/doris/datasource/hive/CachedClient.java [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+import org.apache.hadoop.hive.metastore.api.LockResponse;
+import org.apache.hadoop.hive.metastore.api.NotificationEventResponse;
+import org.apache.hadoop.hive.metastore.api.Partition;
+import org.apache.hadoop.hive.metastore.api.Table;
+import org.apache.hadoop.hive.metastore.api.TableValidWriteIds;
+
+import java.util.List;
+import java.util.Map;
+
+public interface CachedClient extends AutoCloseable {
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/datasource/hive/CachedClient.java:36]:
```suggestion
public interface HMSCachedClient extends AutoCloseable {
```

File: fe/fe-core/src/main/java/org/apache/doris/datasource/hive/CachedClientFactory.java [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    public static CachedClient createCachedClient(PooledHiveMetaStoreClient pooledHiveMetaStoreClient,
+            HiveConf hiveConf, JdbcClientConfig jdbcClientConfig) throws Exception {
+        if (hiveConf != null) {
+            return new IMetaStoreClientCachedClient(hiveConf, pooledHiveMetaStoreClient);
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/datasource/hive/CachedClientFactory.java:31]:
```suggestion
            return new IMetaStoreClientCachedClient(pooledHiveMetaStoreClient, hiveConf);
```
Same order as `HivePostgreSQLCachedClient`

**Comment by BePPPower** [comment]
run buildall

**Comment by BePPPower** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/datasource/property/constants/HMSProperties.java [hunk 5]
```diff
@@ -27,6 +27,7 @@ public class HMSProperties {
     public static final String HIVE_VERSION = "hive.version";
     // required
     public static final String HIVE_METASTORE_URIS = "hive.metastore.uris";
+    public static final String HIVE_META_TYPE = "hive.meta_type";
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/datasource/property/constants/HMSProperties.java:30]:
```suggestion
    public static final String HIVE_METASTORE_TYPE = "hive.metastore.type";
```

**Comment by BePPPower** [comment]
run buildall

**Comment by BePPPower** [comment]
run buildall

**Review by morningman — APPROVED** [review]
LGTM

**Review by zy-kkk — APPROVED** [review]
LGTM

**Pull request merged**